certora/harnesses/GuardianStorageHarness.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.8.12 <0.9.0;
+import {GuardianStorage} from "../../contracts/modules/social_recovery/storage/GuardianStorage.sol";
+
+contract GuardianStorageHarness is GuardianStorage {
+
+    /**
+     * @notice Gets the count of guaridans for a wallet based on the actual linked list.
+     * @dev This is needed for FV to avoid cases where we have values which does not start with SENTINEL.
+     * @param _wallet The target wallet.
+     * @return count of guardians.
+     */
+    function countGuardians(address _wallet) public view returns (uint256 count) {
+        GuardianStorageEntry storage entry = entries[_wallet];
+        if (entry.count == 0 || entry.guardians[SENTINEL_GUARDIANS] == SENTINEL_GUARDIANS || entry.guardians[SENTINEL_GUARDIANS] == address(0)) {
```

Comment:
```suggestion
        if (entry.guardians[SENTINEL_GUARDIANS] == SENTINEL_GUARDIANS || entry.guardians[SENTINEL_GUARDIANS] == address(0)) {
```

I don't think we should be checking `count` at all here. Assuming integrity of the linked list, the check is useless (since count == 0 implies the sentinel guardian conditions that are already in this `if` condition). Since this method is *explicitely* for counting the elements in the linked list, we should not rely on the `count` field.

Reply:
Updated 👍🏾